pandas/io/sql.py

```diff
@@ -1391,7 +1392,17 @@ def to_sql(
             dtype=dtype,
         )
         table.create()
-        table.insert(chunksize, method=method)
+        try:
+            table.insert(chunksize, method=method)
+        except exc.SQLAlchemyError as err:
+            if (
+                str(err.orig)
+                == '(1054, "Unknown column ' + "'inf' in 'field list'" + '")'
+            ):
+                raise ValueError("ProgrammingError: inf can not be used with MySQL")
```

Comment:
I think `inf can not be used with MySQL` is fine for the message.


Also probably good to `raise ValueError(...) from err`